controller/eni_ip_controller.go

```diff
@@ -0,0 +1,267 @@
+// Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package main
+
+import (
+	"os"
+	"strings"
+	"time"
+
+	"github.com/aws/amazon-ecs-cni-plugins/pkg/logger"
+
+	log "github.com/cihub/seelog"
+	"github.com/pkg/errors"
+
+	"k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	lister_v1 "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	kubeletapis "k8s.io/kubernetes/pkg/kubelet/apis"
+
+	"github.com/aws/amazon-vpc-cni-k8s/controller/vpcipresource"
+)
+
+const (
+	defaultLogFilePath = "/host/var/log/aws-routed-eni/controller.log"
+	version            = "0.1.0"
+	instanceTypeLabel  = "beta.kubernetes.io/instance-type"
+)
+
+// ENIIPController is responsible managing VPI IP address resources
+type ENIIPController struct {
+	client     kubernetes.Interface
+	vpcIP      vpcipresource.Interface
+	nodeLister lister_v1.NodeLister
+	queue      workqueue.RateLimitingInterface
+	informer   cache.Controller
+	indexer    cache.Indexer
+	nodeName   string
+	// nodeStoreSynced returns true if the node store has been synced at least once.
+	// Added as a member to the struct to allow injection for testing.
+	nodeStoreSynced cache.InformerSynced
+}
+
+// NewENIIPController creates ENIIPController
+func NewENIIPController(client kubernetes.Interface, vpcIP vpcipresource.Interface, nodeName string) *ENIIPController {
```

Comment:
Take a look at your coverage.  There is a lot of code in this file not covered by test.